docs/probe_hooking_notebook.md

```diff
@@ -0,0 +1,316 @@
+# Probes and Meters: Advanced Hooking Examples for Notebooks
```

Comment:
I think particularly examples 2 and 3 are far too complex and lengthy to include in our official docs. Perhaps we could add example 1 to `instrumentation_examples.md`, although I think we should follow the same protocol of `user_init` rather than show a drastically different approach of using `get_scenario()` and `s.run_attack()` etc. We have separate docs for showing how to run scenarios this way instead of the normal `armory run`

Reply:
removed for now (b282870d) after discussion with @lcadalzo to avoid confusion